src/utilities/utils.js

```diff
@@ -7,23 +7,50 @@ export const cleanData = data => {
       colors: currentPolish.product_colors
     }
   })
-  console.log(freshData)
 
-//reduce through the array
-//inside reduce, filter through the current array's polish brands
-//return only the brands with duplicates in array
-//then, map over this array of duplicates to create a new array of all colors
-//.flat that bb
-//create a new array without duplicates using Set
-//reset the currentPolishes colors to be this new Set
-//push the currentPolish into the accumulator
-//return the reduce
+  const findUnrepeated = array => [...new Set(array)]
 
+  const justBrandNames = freshData.map(element => {
+    return element.brand
+  })
 
-return freshData
+  const uniqueBrands = findUnrepeated(justBrandNames)
 
+  const organizedData = uniqueBrands.map(name => {
+    const newData = {}
+    newData.colors = []
+    freshData.forEach(polObj => {
```

Comment:
I think you can map instead of for each here. Remove 20/21, return the map from the fn and inside of the map return an object with a colors array.